### PR TITLE
refactor: add inference framework

### PR DESCRIPTION
**Pull request by szhao045**
Small refactor to clearly distinguish an inference framework (i.e., an application) from an underlying model (i.e., borzoi) for promoter activity.

### CODE REVIEW

File: ginkgo_ai_client/queries.py [hunk 1]
```diff
@@ -291,19 +293,21 @@ class PromoterActivityQuery(QueryBase):
...
     tissue_of_interest: Dict[str, List[str]]
     source: str
-    model: str = "borzoi-human-fold0"
+    inference_framework: Literal["promoter-0"] = "promoter-0"
+    borzoi_model: Literal["human-fold0"] = "human-fold0"
     query_name: Optional[str] = None
 
     def to_request_params(self) -> Dict:
         # TODO: update the web API so the conversion isn't necessary
+        model_name = f"borzoi-{self.borzoi_model}"
```
Comment (Zulko) [inline ginkgo_ai_client/queries.py]: maybe no need for this variable.

File: test/test_models.py [hunk 2]
```diff
@@ -67,6 +67,23 @@ def test_promoter_activity():
     assert "heart" in response.activity_by_tissue
     assert "liver" in response.activity_by_tissue
 
+def test_promoter_activity_invalid_framework():
```
Comment (Zulko) [inline test/test_models.py]:
```suggestion
def test_promoter_activity_fails_with_invalid_framework_name():
```

**Review by Zulko — COMMENTED** [review]
Small comments. Do you have time to check why the tests are failing for Python3.12 ?

File: ginkgo_ai_client/queries.py (hunk 1 above)
Comment (szhao045) [reply ginkgo_ai_client/queries.py]: thanks!

**Pull request merged**